runtime/compiler/control/CompilationThread.cpp

```diff
@@ -3116,8 +3116,23 @@ void TR::CompilationInfo::stopCompilationThreads()
 
       fprintf(stderr, "Number of compilations per level:\n");
       for (int i = 0; i < (int)numHotnessLevels; i++)
+         {	      
          if (_statsOptLevels[i] > 0)
-            fprintf(stderr, "Level=%d\tnumComp=%d\n", i, _statsOptLevels[i]);
+            {		 
+            fprintf(stderr, "Level=%d\tnumComp=%d\tnumRemoteComp=%d\n", i, _statsOptLevels[i], _statsRemoteOptLevels[i]);
+            totalNumRemoteCompilations += _statsRemoteOptLevels[i];
+            }
+         }
+#if defined(J9VM_OPT_JITSERVER)
+      for (int i = 0; i < JITServer::MessageType_ARRAYSIZE; ++i)
+         {
+         if (JITServerHelpers::serverMsgTypeCount[i] > 0)
+            {
+            totalMsgCount += JITServerHelpers::serverMsgTypeCount[i];
+            }
+         }
+#endif /* defined(J9VM_OPT_JITSERVER) */
+      fprintf(stderr, "Total Number of messages:%d\tAverage number of messages per compilation: %f  \n", totalMsgCount, totalMsgCount/float(totalNumRemoteCompilations));
```

Comment:
This piece of functionality is already done in HookedByTheJIT.cpp JitShutdown():
```
#if defined(J9VM_OPT_JITSERVER)
   static char * isPrintJITServerMsgStats = feGetEnv("TR_PrintJITServerMsgStats");
   if (isPrintJITServerMsgStats && compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
      JITServerHelpers::printJITServerMsgStats(jitConfig);
   static char * isPrintJITServerCHTableStats = feGetEnv("TR_PrintJITServerCHTableStats");
   if (isPrintJITServerCHTableStats)
      JITServerHelpers::printJITServerCHTableStats(jitConfig, compInfo);
#endif
```
All you need to do is modify `JITServerHelpers::printJITServerMsgStats` so that it prints the number of total and average messages. The average can be computed by dividing the number of total messages to the number of messages of type `compilationCode` (every time a remote compilation occurs, the server will send a `compilationCode` message back at the end of the compilation.